gdal/port/cpl_http.cpp

```diff
@@ -1965,6 +1965,36 @@ void *CPLHTTPSetOptions(void *pcurl, const char* pszURL,
     }
 #endif
 
+#if CURL_AT_LEAST_VERSION(7,22,0)
```

Comment:
according to https://github.com/curl/curl/commit/39b9bf60d13ff7cb62a6a031c210d27a32113c4f , the CURL_AT_LEAST_VERSION macro has been introduced only in curl 7.43

Perhaps we should put around the top of this file, after curl include
```
#ifndef CURL_AT_LEAST_VERSION
#define CURL_VERSION_BITS(x,y,z) ((x)<<16|(y)<<8|z)
#define CURL_AT_LEAST_VERSION(x,y,z) \
  (LIBCURL_VERSION_NUM >= CURL_VERSION_BITS(x, y, z))
#endif
```

Reply:
May be replace other `LIBCURL_VERSION_NUM` with `CURL_AT_LEAST_VERSION`?

Reply:
> May be replace other `LIBCURL_VERSION_NUM` with `CURL_AT_LEAST_VERSION`?

good idea. We could assume that the minimum version is 7.19 (super ancient version in RHEL 6 base from what I can see) to get rid of some now useless tests

Reply:
`LIBCURL_VERSION_NUM` macros can be found in the following additional files:
```
gdal/frmts/wms/gdalhttp.cpp
gdal/port/cpl_vsil_curl.cpp
gdal/port/cpl_vsil_curl_class.h
```

To replace`LIBCURL_VERSION_NUM` in these files, it makes sense to move `CURL_AT_LEAST_VERSION` to a common place. But where is it?

Reply:
> To replace`LIBCURL_VERSION_NUM` in these files, it makes sense to move `CURL_AT_LEAST_VERSION` to a common place. But where is it?

maybe create a port/cpl_curl_priv.h that includes <curl/curl.h> and defines CURL_AT_LEAST_VERSION if needed ?